File: C++/3423. Maximum Difference Between Adjacent Elements in a Circular Array.cpp
class Solution {
public:
    int maxAdjacentDistance(vector<int>& n) {
        int m=abs(n[0]-n.back());
        for(int i=0;i<n.size()-1;i++){
            m=max(m,abs(n[i]-n[i+1]));
        }
        return m;
    }
};

File: JAVA/3423. Maximum Difference Between Adjacent Elements in a Circular Array.java
class Solution {
    public int maxAdjacentDistance(int[] n) {
        int m=Math.abs(n[n.length-1]-n[0]);
        for(int i=0;i<n.length-1;i++){
            m=Math.max(m,Math.abs(n[i]-n[i+1]));
        }
        return m;
    }
}